src/components/export-modal/export-modal.jsx

```diff
@@ -141,8 +167,97 @@ class ExportModal extends PureComponent {
     }
   };
 
-  renderExportButton() {
-    if (this.props.status === COMPLETED) {
+  renderExportOptions() {
+    const { isFullCollection } = this.props;
+
+    const queryClassName = classnames({
+      [style('query')]: true,
+      [style('query-is-disabled')]: isFullCollection
+    });
+
+    const queryViewerClassName = classnames({
+      [style('query-viewer')]: true,
+      [style('query-viewer-is-disabled')]: isFullCollection
+    });
+
+    if (this.props.exportProgressStatus === QUERY) {
+      return (
+        <FormGroup controlId="export-collection-option">
+          <div className={style('radio')}>
+            <label className={queryClassName}>
+              <input type="radio"
+                aria-label="Export collection with filters radio button"
+                value="filter"
+                onChange={this.handleExportOptionSelect}
+                checked={!isFullCollection}/>
+              Export query with filters &mdash; {formatNumber(this.props.count)} results (Recommended)
+            </label>
+          </div>
+          <div className={queryViewerClassName}>
+            <QueryViewer
+              query={this.props.query}
+              disabled={isFullCollection}
+              ns={this.props.ns}
+            />
+          </div>
+          <div className={style('radio')}>
+            <label>
+              <input type="radio"
+                aria-label="Export full collection radio button"
+                value="full"
+                onChange={this.handleExportOptionSelect}
+                checked={isFullCollection}/>
+              Export Full Collection
+            </label>
+          </div>
+        </FormGroup>
+      );
+    }
+  }
+
+  renderSelectFields() {
+    if (this.props.exportProgressStatus === FIELDS ) {
```

Comment:
Since `<ExportSelectFields />` has a dependency on `exportProgressStatus` would it be better to move this check be done inside the ExportSelectFields render()?

Reply:
this is good feedback -- will do!